test/player_test.exs

```diff
@@ -138,16 +180,42 @@ defmodule ComputerPlayerTest do
     end
 
     test "returns an empty map", context do
-      assert ComputerPlayer.available_moves(context[:full_board_state]) == %{}
+      assert Player.available_moves(context[:full_board_state]) == %{}
     end
 
     test "returns empty error message", context do
-      assert ComputerPlayer.select_move(context[:full_board_state]) ==
+      assert Player.select_move(context[:full_board_state]) ==
                {:error, "empty error"}
     end
+  end
 
-    test "returns state when board is full", context do
-      assert ComputerPlayer.make_move(context[:full_board_state]) == context[:full_board_state]
-    end
+  test "rejects invalid input until user selects \"X\"" do
+    Helpers.Stack.setup(["foo", "10\n", "1\n", "4\n", "x\n"])
+
+    mock_deps = %{
+      validation: MockValidation,
+      messages: MockMessages,
+      io: MockTTT.IO,
+      human_player: MockHumanPlayer
+    }
+
+    assert Player.choose_token(mock_deps) == "X"
+
+    Helpers.Stack.teardown()
+  end
+
+  test "rejects invalid input until user selects \"O\"" do
+    Helpers.Stack.setup(["foo", "10\n", "1\n", "4\n", "o\n"])
```

Comment:
Not really feedback, so forgive me for that, but I'm curious:  How do you like this `Helpers` pattern?

Reply:
I've found it to be pretty useful overall, especially for testing validation functions which make recurring calls to get input until something valid is passed in.